batch/batch/worker/worker.py

```diff
@@ -1086,13 +1097,30 @@ async def run(self):
                         headers=self.headers)
                 log.info('deactivated')
         finally:
-            log.info('shutting down')
-            if site:
-                await site.stop()
-                log.info('stopped site')
-            if app_runner:
-                await app_runner.cleanup()
-                log.info('cleaned up app runner')
+            await self.cleanup()
+            self.shutdown()
+
+    async def kill(self, request):
+        body = await request.json()
+
+        expected_name = body['name']
+        if expected_name != NAME:
+            return web.HTTPForbidden()
```

Comment:
If we're concerned enough that we could have the wrong IP for an instance in this case, shouldn't we verify the name on every request?

Reply:
I'll leave this for a separate PR.